plugin/evm/gossiper.go

```diff
@@ -405,18 +413,27 @@ func (n *pushGossiper) GossipEthTxs(txs []*types.Transaction) error {
 
 // GossipHandler handles incoming gossip messages
 type GossipHandler struct {
-	vm            *VM
-	atomicMempool *Mempool
-	txPool        *txpool.TxPool
-	stats         GossipReceivedStats
+	vm              *VM
+	atomicMempool   *Mempool
+	txPool          *txpool.TxPool
+	atomicTxMempool gossip.Mempool[*GossipAtomicTx]
+	ethTxMempool    gossip.Mempool[*GossipEthTx]
```

Comment:
these sound a bit similar to txPool and atomicMempool. I think we should either emphasize `pullGossipMempool` or `bloomFilterMempools` in the package/interface/struct/variable names. Sorry if this was a bit generic comment :D

Reply:
Yeah these variables only exist because the compiler can't figure out that these satisfy the interface type for some reason. But yeah I'm down to rename this.